### PR TITLE
Refactor/add missing api methods

### PR DESCRIPTION
**Pull request by Eazybright**
The following endpoints have the added to the SDK:

- delete_subscriber_credentials
- get_blueprint
- group_blueprints_by_category
- set_integration_as_primary
- delete_subscriber_credentials
- mark_all_subscriber_messages
- provider_oauth_redirect
- chat_oauth
- delete_topic
- subscriber_topic

### CODE REVIEW

**Comment by sumitsaurabh927** [comment]
Please address the comments @Eazybright

**Comment by Eazybright** [comment]
> Please address the comments @Eazybright

I can't find any comments here @sumitsaurabh927

**Comment by sumitsaurabh927** [comment]
@Eazybright please go through each commit 🙂

**Comment by Eazybright** [comment]
> @Eazybright please go through each commit 🙂

I've search al the commits and can't find any comments still. Can you share the url of one of the comments for easy navigation?

**Comment by Eazybright** [comment]
> > @Eazybright please go through each commit 🙂
> 
> I've search al the commits and can't find any comments still. Can you share the url of one of the comments for easy navigation?

@sumitsaurabh927

**Comment by sumitsaurabh927** [comment]
> > > @Eazybright please go through each commit 🙂
> > 
> > 
> > I've search al the commits and can't find any comments still. Can you share the url of one of the comments for easy navigation?
> 
> @sumitsaurabh927

You can find an example [here](https://github.com/novuhq/novu-ruby/pull/40/commits/099e8768234525d839d4e200e09f9e6f1b8f4a6c). Go through other commits similarly.
SCR-20231006-nkcb

**Comment by Eazybright** [comment]
Thanks. But comments are not showing on my end.

**Comment by unicodeveloper** [comment]
@sumitsaurabh927 I can't see any comments as well

**Comment by unicodeveloper** [comment]
I can see they are Pending on your end. It means you need to finish reviewing it and publish the comments so that @Eazybright and anyone looking at this PR can see the comments.

File: spec/blueprints_spec.rb
[the remark this answers is not in the document]
Comment (Eazybright) [reply spec/blueprints_spec.rb:1]: What do you mean by placeholder text?

[the remark this answers is not in the document]
Comment (sumitsaurabh927) [reply spec/blueprints_spec.rb:1]: Use something like `<Template_Id>` instead of actual data

**Review by unicodeveloper — APPROVED** [review]

File: spec/blueprints_spec.rb
[the remark this answers is not in the document]
Comment (Eazybright) [reply spec/blueprints_spec.rb:1]: Noted. PR updated

**Pull request merged**